src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/SupplementaryDatasetService.java

```diff
@@ -28,7 +28,17 @@ public void addSupplementaryDatasetEntity(SupplementaryDatasetDTO supplementaryD
     CollectionExercise collectionExercise =
         collectionExerciseService.findCollectionExercise(
             supplementaryDatasetDTO.getSurveyId(), supplementaryDatasetDTO.getPeriodId());
-
+    if (collectionExercise == null) {
+      log.error(
```

Comment:
As exceptions are logged to stderr by default, we probably don't need this line as it will simply duplicate the throw message below. Now we've added the `survey_id` and `period_id` into the exception we should have all the info we need to alert on.